### ISSUE TITLE
Replace CompareFamily with fontbakery

### ISSUE THREAD

**Issue by miguelsousa** [adobe-type-tools/afdko#501]
Write a [fontbakery specification](https://github.com/googlefonts/fontbakery/blob/master/docs/writing-specifications.md) that covers the tests (family and single face) performed by CompareFamily.

**Comment by miguelsousa** [comment]
Address #213 during the conversion.

**Comment by felipesanches** [comment]
@miguelsousa I am happy to hear that you are interested in porting these checks into FontBakery. I have written all the boilerplate for that in order to help you in this effort. Please take a look at https://github.com/googlefonts/fontbakery/issues/1992 and let me know what you think

**Comment by felipesanches** [comment]
In particular, also take a look at the pull request where the boilerplate code is: https://github.com/googlefonts/fontbakery/pull/2064

It also contains an example of a port of one of your checks.

**Comment by miguelsousa** [comment]
@felipesanches thanks for the head start. @cjchapman will be leading the effort with input from me. Many of the comparefamily tests are either obsolete or need to be adjusted, so I'm glad you didn't go too far. I honestly dislike many of the code-style options used, but that's a minor thing that we can easily change.

**Comment by felipesanches** [comment]
yeah, feel free to fetch the commits from that pull request and work on top of them. Then later you can open a new pull request containing your work in progress. The sooner you do so, the better, as I can help you along the way.

I completely understand the feeling of disliking to code-style of an aging code-base. Fontbakery itself was very messy in the past, but we had a big refactor effort some time ago and we finally reached the current state, which I like a lot: each check being a small self-contained and easier to read piece of code.

I think I will still sketch the code-test boilerplate for the afdko spec. And then I'll leave the rest for you and @cjchapman 

But feel free to ask me questions whenever you need further help!

Happy hacking,
Felipe Sanches

**Comment by arialcrime** [comment]
Curious to find out what the current state of this endeavor is. Do you already roughly know when (the good parts of) CompareFamily will be fully integrated into Fontbakery?

**Comment by josh-hadley** [comment]
There has been some progress; a few have already been ported over into either the adobefonts, opentype, or universal profiles. We've done some additional work, prioritizing checks that are unique to compareFamily that haven't made it into fontbakery yet, and will continue working on getting those ported over.

If you have strong ideas about what "the good parts" of compareFamily are that need to make it over, we'd love to get help with that!

**Comment by arialcrime** [comment]
Oh that’s great news, really looking forward to it!

With “the good parts” I was merely referring to everything that passes the bar of not being obsolete. I like all of them equally. ;-)

**Comment by felipesanches** [comment]
I'm happy to hear that as well :-)
I'll be glad to review and merge pull requests with these new additions to the above mentioned profiles.

**Comment by felipesanches** [comment]
I'm happy to hear that as well :-)
I'll be glad to review and merge pull requests with these new additions to the above mentioned profiles.
